pkg/controller/controller.go

```diff
@@ -247,12 +251,17 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	if err != nil {
 		return nil, err
 	}
-	// create random share name
-	share := fmt.Sprintf("%s-%s", p.volumeNamePrefix, strings.Replace(string(uuid.NewUUID()), "-", "", -1)[0:p.volumeNameUUIDLength])
+	// create persistent name based on a volumeNamePrefix and volumeNameUUIDLength
+	// of PVC's UID
+	pvcUID := options.PVC.ObjectMeta.UID
+	if len(pvcUID) == 0 {
+		return nil, fmt.Errorf("corrupted PVC object, it is missing UID")
+	}
+	share := fmt.Sprintf("%s-%s", p.volumeNamePrefix, strings.Replace(string(pvcUID), "-", "", -1)[0:p.volumeNameUUIDLength])
```

Comment:
Is pvcUID guaranteed to be unique across namespaces? If so, that is a pretty cool trick, and should be suggested to https://github.com/kubernetes/community/pull/1790 as well

Reply:
Well it is not 100% guaranteed since there is a super small opportunity of the collision, but I think since the chances are so small, so imho this way should be acceptable.

Reply:
We should align with Kubernetes

Reply:
done